stdlib/public/core/RangeReplaceableCollection.swift.gyb

```diff
@@ -1142,11 +1142,11 @@ extension RangeReplaceableCollection
 public func + <C : RangeReplaceableCollection, S : Sequence>(lhs: C, rhs: S) -> C
   where S.Iterator.Element == C.Iterator.Element {
 
-  var lhs = lhs
```

Comment:
I don't think this change is necessarily a good idea.  When ARC is doing its job and the LHS parameter is otherwise unused, reserveCapacity should find itself with a unique reference, and if there's already enough storage, there should be no reallocation.  IMO that optimization is more important than supporting + on collections with reference semantics.